### DISCUSSION TITLE
Shorts/Snippets/Tips/Tricks: A proposal for a new type of content on the Dev Blog

### DISCUSSION THREAD

**Discussion by justintadlock** [WordPress/developer-blog-content#314] [Topic Ideas]
One of the areas that the Dev Blog excels at is with long-form, deep-diving tutorials on big subjects. However, we're missing a solid short-form of content that I don't think we've truly explored at this point. We don't necessarily have word minimums/maximums, but I think there's a general thought that everything has to be some epic undertaking.

I'd like to propose a new category titled "Shorts" or "Snippets" or some other comparable name. They would follow a format similar to this:

```md
Quick intro paragraph of a problem.

Optional second paragraph describing a solution.

[-- Code snippet to solve the problem. --]

Maybe a closing paragraph with further explanation.
```

The goal is to keep these at less than X number of words (400 words or so?) and describe a quick solution. I believe this would be more inviting to some potential authors, and it wouldn't need to feel like such an epic undertaking to add useful info on the blog.

For me, the goal is to share tips and tricks that aren't usually covered in the documentation. I have tons of snippets like these that'll never see the light of day, but they're really not worth writing a full-on tutorial about.

Side note: it'd be cool to have a featured image design that is specific to these types of posts too.

## Example Short

Title: How to add the Core button element class to the login/out block

A description of the problem that it doesn't have the `.wp-element-button` class and can't be styled through `theme.json`.  Then showcase a short snippet of using the HTML API to add a class...

```php
add_filter( 'render_block_core/loginout', 'themeslug_render_loginout', 10, 2 );

function themeslug_render_loginout(string $content, array $block): string
{
	if (
		empty($block['attrs']['displayLoginAsForm'])
		|| is_user_logged_in()
	) {
		return $content;
	}

	$processor = new WP_HTML_Tag_Processor($content);
	
	while ($processor->next_tag()) {
		if (
			'INPUT' === $processor->get_tag()
			&& 'wp-submit' === $processor->get_attribute('name')
		) {
			$processor->add_class(wp_theme_get_element_class_name('button'));
			break;
		}
	}

	return $processor->get_updated_html();
}
```

**Comment by mikemcalister** [comment]
I love this idea and I think it can have an even better chance at success if posted on X with regularity. 

Does the team have an app like Typefully where you can collaborate on X content and auto repost it? Lots of content gets lost if shared only once. Automating that bit will ensure it continues to get shared and keeps it in builders periphery.

**Reply by justintadlock** [reply]
Right now, it's a very manual process for sharing via WordPress socials, at least from our end. It may be possible to schedule it posting more often than once (I'd have to ask).

**Reply by mikemcalister** [reply]
Yeah I hear that. What I often hear from folks (and experience it myself) is content kinda gets lost in .org. It may only be reaching a small slice of our community.

So whatever we can do to elevate it out of the site and point people back would be great. 

I love the effort to create more unique and useful content (I'd love to do something on .org myself), I just want everyone's efforts to be seen by the masses.

**Comment by eirichmond** [comment]
@justintadlock I think this is a great idea! A holistic solution to any project is made up of several micro-solutions.

I often find snippets of useful information in long-form tutorials, but the full tutorials are usually too opinionated for most use cases. 

This reminds me of an old project of mine where I reached out to developers to submit snippets did you submit one @justintadlock? many of these snippets I still refer to today. Bite-sized shorts, snippets, tips, and tricks would be incredibly useful in my opinion.

**Reply by justintadlock** [reply]
I probably did submit at one point. What was the name of the site? It may have been one I've been thinking about it all this but couldn't remember the name.

**Reply by eirichmond** [reply]
It was WP Snippets Til Xmas, a cheesy advent calendar for folks running up to the big day 🎅🏼

**Comment by colorful-tones** [comment]
I'm not opposed to the idea and think it could encourage more authors to contribute. However, I would like to clarify the maintenance of existing and future posts (regardless of format). Does the Developer Blog have a policy or standard around whether an original author needs to maintain their posts, including responding to comments, updating code shared to adhere to API changes, etc.?

My concern is that snippets are shared, they become canonical amongst the community, but they're never updated, and eventually, they become irrelevant, potentially impacting the reputation of the Developer Blog.

As far as naming: shorts, snippets, tips, tricks - my vote is for snippet.

> Side note: it'd be cool to have a featured image design that is specific to these types of posts too.

Some type of visual differentiation would be necessary to allow site visitors to glance and garner what the different types of content offered. 👍

**Reply by justintadlock** [reply]
> However, I would like to clarify the maintenance of existing and future posts (regardless of format). Does the Developer Blog have a policy or standard around whether an original author needs to maintain their posts, including responding to comments, updating code shared to adhere to API changes, etc.?
>
> My concern is that snippets are shared, they become canonical amongst the community, but they're never updated, and eventually, they become irrelevant, potentially impacting the reputation of the Developer Blog.

Right now, a few of us have been manually updating a handful of errors behind the scenes, but we don't have any formal process right now. I think this is generally OK: iterate on what we need over time and build up ideas for solutions.

I encourage bringing this up for the next meeting for discussion, though, definitely.

There's also the case to be made that this content isn't meant to be evergreen (that's what the docs are for). We can always write new tutorials to replace things that are outdated.

I'm personally split a little bit in both directions. For minor things, small changes to the code in posts would generally be OK. For things that are _really_ outdated, I'd be in favor of a notice on the post (one that would link to a newer post if one is written).

> As far as naming: shorts, snippets, tips, tricks - my vote is for snippet.

That's what I'm thinking too. 👍

**Comment by justintadlock** [comment]
I've written a Snippet-style post using one of the pre-approved topics here: https://github.com/WordPress/developer-blog-content/issues/307

Google Doc here: https://docs.google.com/document/d/1S4aWXZ-Q3GUklA82XAi4fq8ZdtJBseAYl9I_zeSAeVI/edit?usp=sharing

**Comment by unscripted** [comment]
I absolutely love this idea!

Naming idea: The gist. 

Then, all code snippets can be linked to a GitHub gist, where users can star them, subscribe and fork them as needed.

**Reply by justintadlock** [reply]
I like the idea, but I don't think we can use Gist because they are tied to personal accounts. If I'm not mistaken, we can't assign Gists to the Org account (could be wrong, though).

At the very least, I'd like them tracked in a repo.

**Reply by unscripted** [reply]
Looks like you're correct. I just tried to create one as the org but it switched back to my profile.

Count me as a +1 for Snippets.

**Reply by marybaum** [reply]
I do love the name The Gist for the editorial category, even if we can't use gists as the snippets. I've been wondering if this category — a column, perhaps, speaking from an editorial standpoint? — could have a name and some content that goes beyond code snippets to other sorts of design, devops-ish and other tips we run across.

Something like Snippets, Tips, and Tools — or Quick Tips and Fast Fixes — almost tried to use Fast Breaks, as if I'm not the mother-in-law of someone in college basketball (Go Huskies!) I can think a little harder and come up with something better, if yall like.

Also let me know if you'd like some graphic help with this. Been working on my [icons and bugs lately—even a wapuu.](https://www.supportinclusionintech.com/)

**Comment by sethrubenstein** [comment]
Love this idea. I've been struggling to finish longer form content for the dev blog because of time constraints; shorter form quick content would be easier to contribute to.

**Comment by justintadlock** [comment]
I think it's safe to close this discussion now. Thanks for your feedback, everyone!

The first snippet is live on the site now: https://developer.wordpress.org/news/snippets/conditionally-unregister-patterns/

We're still working through some of the implementation details here: https://github.com/WordPress/wporg-developer-blog/pull/70

And we'll need to make some decisions around that in the next Editorial Group meeting.
